lib/std/os/windows.zig

```diff
@@ -215,24 +215,28 @@ pub fn CreateEventExW(attributes: ?*SECURITY_ATTRIBUTES, nameW: [*:0]const u16,
     }
 }
 
-pub const FsControlFileError = error{Unexpected};
+pub const DeviceIoControlError = error{Unexpected};
 
-// TODO work out if we need to expose other arguments to the underlying
-// NtFsControlFile syscall
-pub fn FsControlFile(
+/// A Zig wrapper around `NtDeviceIoControlFile` and `NtFsControlFile` syscalls.
+/// It implements similar behavior to `DeviceIoControl` and is meant to serve
+/// as a direct substitute for that call.
+/// TODO work out if we need to expose other arguments to the underlying syscalls.
+pub fn DeviceIoControl(
     h: HANDLE,
-    fsControlCode: ULONG,
+    ioControlCode: ULONG,
     in: ?[]const u8,
     out: ?[]u8,
-) FsControlFileError!void {
+) DeviceIoControlError!void {
+    // Logic from: https://doxygen.reactos.org/d3/d74/deviceio_8c.html
+    const syscall = if ((ioControlCode >> 16) == FILE_DEVICE_FILE_SYSTEM) ntdll.NtFsControlFile else ntdll.NtDeviceIoControlFile;
```

Comment:
does this end up using a function pointer instead of a normal call?

Reply:
yes

Reply:
Would it be better to have two function call code paths then?

Reply:
yes

Reply:
That's a good point. I'll provide a quickfix ASAP before we forget to apply this!